test/performance/shmem_perf_suite/shmem_bw_nb_atomics_perf.c

```diff
@@ -0,0 +1,283 @@
+                                (TYPE *)(metric_info->src), dest, ONE, dest);  \
+                            shmem_quiet();                                     \
+                    }                                                          \
+                }                                                              \
+                shmem_barrier_all();                                           \
+                if(snode) {                                                    \
+                    start = perf_shmemx_wtime();                               \
+                    for(i = 0; i < metric_info->trials; i++) {                 \
+                        for(j = 0; j < metric_info->window_size; j++)          \
+                            shmemx_##NAME##_atomic_compare_swap_nbi((TYPE *) &fetch,    \
+                                (TYPE *)(metric_info->src), dest, ONE, dest);  \
+                            shmem_quiet();                                     \
+                    }                                                          \
+                    end = perf_shmemx_wtime();                                 \
+                }                                                              \
+            break;                                                             \
+            default:                                                           \
+                fprintf(stderr, "Error %d not a valid op case                  \
+                                                for atomics\n", op);           \
+            break;                                                             \
+        }                                                                      \
+        if(snode) {                                                            \
+            calc_and_print_results(end, start, len, metric_info);             \
+        }                                                                      \
+    } while(0)
+
+#define NUM_INC 100
+
+static const char * atomic_op_names [] = { "fetch", "cswap", "swap", "finc", 
+                                           "fadd", "fand", "for", 
+                                           "fxor" };
+
+
+static inline void bw_set_metric_info_len(perf_metrics_t * const metric_info)
+{
+    unsigned int atomic_sizes[ATOMICS_N_DTs] = {sizeof(unsigned int), sizeof(unsigned long),
+                                        sizeof(unsigned long long)};
+    metric_info->b_type = UNI_DIR;
+    int snode = streaming_node(metric_info);
+    atomic_op_type op_type = OP_FETCH;
+
+    for(op_type = OP_FETCH; op_type <= OP_FXOR; op_type++) {
+        if(metric_info->my_node == 0) { 
+            printf("\nshmemx_%s\n", atomic_op_names[op_type]);
+            printf("-----------\n");
+        }
+
+        metric_info->start_len = atomic_sizes[0];
+        metric_info->max_len = atomic_sizes[0];
+        metric_info->size_inc = NUM_INC;
```

Comment:
What's this for?  It doesn't seem to get used...

Reply:
Good question :). There are some messy codings here.. I will clean up these.